demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1051 @@
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONWriter_t xWriter;
+    int32_t lBytesWritten;
+
+    xResult = AzureIoTJSONWriter_Init( &xWriter, ucPropertyPayloadBuffer, sizeof( ucPropertyPayloadBuffer ) );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendBeginObject( &xWriter );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTHubClientProperties_BuilderBeginResponseStatus( &xAzureIoTHubClient,
+                                                                      &xWriter,
+                                                                      ( const uint8_t * ) sampleazureiotPROPERTY_TARGET_TEMPERATURE_TEXT,
+                                                                      strlen( sampleazureiotPROPERTY_TARGET_TEMPERATURE_TEXT ),
+                                                                      sampleazureiotPROPERTY_STATUS_SUCCESS,
+                                                                      ulVersion,
+                                                                      ( const uint8_t * ) sampleazureiotPROPERTY_SUCCESS,
+                                                                      strlen( sampleazureiotPROPERTY_SUCCESS ) );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendDouble( &xWriter, xUpdatedTemperature, sampleazureiotDOUBLE_DECIMAL_PLACE_DIGITS );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTHubClientProperties_BuilderEndResponseStatus( &xAzureIoTHubClient,
+                                                                    &xWriter );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendEndObject( &xWriter );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    lBytesWritten = AzureIoTJSONWriter_GetBytesUsed( &xWriter );
+    configASSERT( lBytesWritten > 0 );
+
+    LogDebug( ( "Sending acknowledged writable property. Payload: %.*s", lBytesWritten, ucPropertyPayloadBuffer ) );
+    xResult = AzureIoTHubClient_SendPropertiesReported( &xAzureIoTHubClient, ucPropertyPayloadBuffer, lBytesWritten, NULL );
+
+    if( xResult != eAzureIoTSuccess )
+    {
+        LogError( ( "There was an error sending the reported properties: 0x%08x", xResult ) );
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Property mesage callback handler
+ */
+static void prvHandleProperties( AzureIoTHubClientPropertiesResponse_t * pxMessage,
+                                 void * pvContext )
+{
+    AzureIoTResult_t xResult;
+    double xIncomingTemperature;
+    uint32_t ulVersion;
+    bool xWasMaxTemperatureChanged = false;
+
+    ( void ) pvContext;
+
+    LogDebug( ( "Property document payload : %.*s \r\n",
+                pxMessage->ulPayloadLength,
+                pxMessage->pvMessagePayload ) );
+
+    switch( pxMessage->xMessageType )
+    {
+        case eAzureIoTHubPropertiesGetMessage:
```

Comment:
Are we sure this is `eAzureIoTHubClientPropertyWritable`?  This is same value we pass in block below with `eAzureIoTHubPropertiesWritablePropertyMessage` (where it's correct).  I think this is `eAzureIoTHubPropertiesGetMessage` ??

Reply:
I think this is right. This mirrors this section of the embedded:
https://github.com/Azure/azure-sdk-for-c/blob/97cde2c6f5d4fc0b411c8ea9b61192f6f9424a0d/sdk/samples/iot/paho_iot_pnp_component_sample.c#L529-L554

There's a chance we need some naming massaging here because these are the two current types we might be confusing here:

`AzureIoTHubMessageType_t`
https://github.com/Azure/azure-iot-middleware-freertos/blob/99443168dc24d431edc44cd253635996d429e433/source/include/azure_iot_hub_client.h#L61-L68

 `AzureIoTHubClientPropertyType_t`
https://github.com/Azure/azure-iot-middleware-freertos/blob/99443168dc24d431edc44cd253635996d429e433/source/include/azure_iot_hub_client_properties.h#L221-L227

In the embedded we also use the "writeable" property type as well.

Reply:
We can discuss whether there's some name improvements regardless?

Reply:
OK I think I see - the `pxMessage->xMessageType` is indeed making it down to lower layer when `prvProcessProperties` passes in `pxMessage->xMessageType` to `AzureIoTHubClientProperties_GetNextComponentProperty`.

Naming is ongoing discussion I'm all for having but not right before pub preview.